internal/memory/store.go

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Lekko Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memory
+
+import (
+	"fmt"
+	"sync"
+
+	backendv1beta1 "buf.build/gen/go/lekkodev/cli/protocolbuffers/go/lekko/backend/v1beta1"
+	featurev1beta1 "buf.build/gen/go/lekkodev/cli/protocolbuffers/go/lekko/feature/v1beta1"
+	"github.com/lekkodev/go-sdk/pkg/eval"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	ErrConfigNotFound error = fmt.Errorf("config not found")
+)
+
+func newStore() *store {
+	return &store{
+		configs: make(map[string]map[string]configData),
+	}
+}
+
+type configData struct {
+	config    *featurev1beta1.Feature
+	configSHA string
+}
+
+// store implements an in-memory store for configurations. It stores
+// the commit sha of the contents. It supports atomically updating
```

Comment:
I've see two style of implementing atomic storage in Go: mutext and putting state into a goroutine and communicate through channels. Do you know which one is faster?

Reply:
both are valid approaches. for this implementation, I find the mutex approach simpler and less error-prone. 

we have 1 writer and `n` readers, so the mutex approach is quite performant here.